pymc/distributions/timeseries.py

```diff
@@ -237,64 +298,27 @@ def logp(self, x):
         -------
         TensorVariable
         """
-        if x.ndim > 0:
-            x_im1 = x[:-1]
-            x_i = x[1:]
-            mu, sigma = self._mu_and_sigma(self.mu, self.sigma)
-            innov_like = Normal.dist(mu=x_im1 + mu, sigma=sigma).logp(x_i)
-            return self.init.logp(x[0]) + at.sum(innov_like)
-        return self.init.logp(x)
 
-    def random(self, point=None, size=None):
-        """Draw random values from GaussianRandomWalk.
+        def normal_logp(value, mu, sigma):
+            logp = (
+                -0.5 * at.pow((value - mu) / sigma, 2)
+                - at.log(at.sqrt(2.0 * np.pi))
+                - at.log(sigma)
+            )
+            return logp
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
+        # Create logp calculation graph the inital time point
+        init_logp = normal_logp(value[0] - init, mu, sigma)
```

Comment:
Your shape issues probably come from here. You will end up trying to concatenate the scalar init logp with the vector innov logp. If you instead select the initial value like this, it should work:

```suggestion
        init_logp = normal_logp(value[:1] - init, mu, sigma)
```

Alternatively you can use the strategy we use in aeppl for cumsum "SymbolicDists", where we first concatenate the initial value with the vector of differences and only then call the logp: https://github.com/aesara-devs/aeppl/blob/d09a01d8f165c709be0b73b51d5a8725d91d7d94/aeppl/cumsum.py#L20

(Note that code is a bit more complicated because we handle cumsum across any axis, you only need to worry about `axis=-1` since thats what the GRW Op uses).

Reply:
Thanks!